src/StackExchange.Redis/Interfaces/IDatabase.cs

```diff
@@ -2202,6 +2202,18 @@ public interface IDatabase : IRedis, IDatabaseAsync
         /// <remarks>https://redis.io/commands/msetnx</remarks>
         bool StringSet(KeyValuePair<RedisKey, RedisValue>[] values, When when = When.Always, CommandFlags flags = CommandFlags.None);
 
+        /// <summary>
+        /// Atomically sets key to value and returns the old value stored at key.
+        /// </summary>
+        /// <param name="key">The key of the string.</param>
+        /// <param name="value">The value to set.</param>
+        /// <param name="expiry">The expiry to set.</param>
+        /// <param name="when">Which condition to set the value under (defaults to always).</param>
+        /// <param name="flags">The flags to use for this operation.</param>
+        /// <returns>The old value stored at key, or nil when key did not exist.</returns>
+        /// <remarks>https://redis.io/commands/set</remarks>
```

Comment:
```suggestion
        /// <summary>
        /// Atomically sets key to value and returns the previous value (if any) stored at <paramref name="key"/>.
        /// </summary>
        /// <param name="key">The key of the string.</param>
        /// <param name="value">The value to set.</param>
        /// <param name="expiry">The expiry to set.</param>
        /// <param name="when">Which condition to set the value under (defaults to <see cref="When.Always"/>).</param>
        /// <param name="flags">The flags to use for this operation.</param>
        /// <returns>The old value stored at <paramref name="key"/>, or nil when key did not exist.</returns>
        /// <remarks>https://redis.io/commands/getset</remarks>
```

Reply:
Thank you very much for checking this out!

I did the requested changes with one exception, since I think it might be due a slight confusion. I have left the link in the remarks as <https://redis.io/commands/set>, since that is the link that points to the `SET` method which in version **6.2.0** has the `GET` option — this is what this method uses. On the other hand, <https://redis.io/commands/getset> points to the `GETSET` method which has been deprecated in **6.2.0**.

Please let me know if I misunderstood your reasons / intent, and I'll happily change the link as well!

Reply:
Oh no you're absolutely right - I think we should probably add that distinction to the remarks though!

Reply:
Looking good, thanks!